packages/teams-js/src/public/authentication.ts

```diff
@@ -202,6 +195,12 @@ export namespace authentication {
     });
   }
 
+  function throwIfStringNotValidHttpsUrl(link: string): void {
+    if (!isValidHttpsURL(new URL(link))) {
+      throw new Error(`${ErrorCode.INVALID_ARGUMENTS}: ${link} must be valid https URL`);
```

Comment:
What if the URL is a https URL but while casting the string to new URL() failed?. Do we want to throw that error.
This can happen when the input is an encoded URI and the new URL(endcodedURL) fails.

Reply:
Can you give me an example of the type of URL you are talking about? Not sure I follow.

Reply:
ex: 'https%3A%2F%2Faccounts.microsoft.com%2F' is a valid url but it will fail the casting to new URL () because it is encoded.

Reply:
Talked about this with @ndangudubiyyam offline. This function is in a bit of a weird state because it allows strings to be passed in as URLs (strong typing is our friend and we should use it whenever we can).

What I have here (no use of decode) is actually correct relative to how this function is already working. Within the function there is code like this:

![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/18446904/b10e82b2-f683-4d0f-af73-76f8661b5536)

This code is intended to turn relative URLs into absolute ones by appending them to the end of current path (which is something you get for free when inserting a string into an <a> tag. That means that _before this change_ the type of URL that @ndangudubiyyam is describing would have already not been legal to pass in (an url encoded absolute URL) because it would have been identified as a relative URL and appended to the current path, giving you something like:

`https://accounts.microsoft.com/https%3A%2F%2Faccounts.microsoft.com%2F`

However, a url encoded _relative_ url would have worked just fine. If the url string `hello%20world` was passed into this function you'd get something like:

`https://accounts.microsoft.com/hello%20world`

Which is a legal URL!

I actually don't think I would want full URLs that are encoded to be allowed to be passed in here anyways, so I'll make sure we maintain the current behavior.